src/fugw/mappings/dense.py

```diff
@@ -125,6 +143,23 @@ def fit(
         Ds = _make_tensor(source_geometry, device=device)
         Dt = _make_tensor(target_geometry, device=device)
 
+        # Do the same for validation data if it was provided
+        if source_features_val is not None and target_features_val is not None:
+            Fs_val = _make_tensor(source_features_val.T, device=device)
+            Ft_val = _make_tensor(target_features_val.T, device=device)
+            F_val = torch.cdist(Fs_val, Ft_val, p=2) ** 2
+
```

Comment:
I like it better when `if, elif, else` statements are grouped together, but that's just a personal preference ;)